[CS2113-F10-3] Meal360

jaredoong opened the pull request:
Meal 360 is a software for meals preparation for the week via Command Line Interface (CLI). Users would be able to use it to streamline the process of creating using recipes.

gurmankalkat committed:
match actual & expected output

gurmankalkat committed:
fix json formatting

gurmankalkat committed:
fix actual vs expected bug

gurmankalkat committed:
fix formatting of expected.txt

TopGun2001 committed:
Changes: Accepting ingredient name in 1 line

gurmankalkat committed:
add back newline

TopGun2001 committed:
runtest works

gurmankalkat committed:
add another newline

gurmankalkat committed:
copy in new expected file

gurmankalkat committed:
add back file

gurmankalkat committed:
add newline

gurmankalkat committed:
delete test

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-withUpstream

jaredoong committed:
Merge pull request #50 from gurmankalkat/master

clear weekly plan & view weekly ingredients. Will fix the failing test on my end with a separate pull request

jaredoong committed:
Merge branch 'master' into master

jaredoong committed:
Merge pull request #45 from notbingsu/master

Add Ingredient and IngredientList classes, will update expected.txt and input.txt accordingly

TopGun2001 committed:
removed assert statement in main

jaredoong committed:
Fix merge conflicts

jaredoong committed:
Merge branch 'TopGun2001-branch-level-1'

jaredoong committed:
Merge branch 'master' into branch-level-1

jaredoong committed:
Merge pull request #51 from TopGun2001/branch-level-1

Will fix expected and input.txt accordingly

jaredoong committed:
Fix merge conflicts

Also fix input and expected.txt, and integrated team code.

jaredoong committed:
Edit sourcing of database

Remove print statements and change to use universal path separator

JuneNita committed:
Merge pull request #48 from jaredoong/branch-weeklyplanner

Update weekly planner add and delete functionality

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-withUpstream

jaredoong committed:
Fix merge conflicts

JuneNita committed:
update DG (add 'list' and 'tag')

TopGun2001 committed:
Merge pull request #52 from JuneNita/JuneNita-withUpstream

add DG for 'list' and 'tag' implementation

TopGun2001 committed:
no message

TopGun2001 committed:
Merge https://github.com/AY2223S2-CS2113-F10-3/tp into branch-level-1

gurmankalkat committed:
update DG & some comments

TopGun2001 committed:
DG

TopGun2001 committed:
Merge pull request #54 from TopGun2001/branch-level-1

Branch level 1

gurmankalkat committed:
Merge branch 'master' into master

gurmankalkat committed:
Merge pull request #53 from gurmankalkat/master

update DG & some comments

JuneNita committed:
add list and tag UML in DG

TopGun2001 committed:
Merge pull request #56 from JuneNita/JuneNita-withUpstream

add list and tag UML in DG

JuneNita committed:
edit the UML pic format

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-withUpstream

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-withUpstream

jaredoong committed:
Merge branch 'master' of github.com:AY2223S2-CS2113-F10-3/tp

jaredoong committed:
Update Database

Now includes database for ingredients and weeklyplan

JuneNita committed:
add 'random function' and make it more readable

JuneNita committed:
update error message

JuneNita committed:
update UserGuide

JuneNita committed:
update develop guide

jaredoong committed:
Merge pull request #60 from JuneNita/JuneNita-withUpstream

add 'random' function, update UG and DG

jaredoong committed:
Merge branch 'master' of github.com:AY2223S2-CS2113-F10-3/tp into enhancements

notbingsu committed:
no message

jaredoong committed:
Refactor weeklyplan code

Refactored duplicated code. User can now select the quantity of recipes to be deleted from the recipe. Updated user guide accordingly.

jaredoong committed:
Add commands

User can now add, delete and view their ingredients

notbingsu committed:
Merge branch 'master' of https://github.com/notbingsu/tp

jaredoong committed:
Fix JUnit test

notbingsu committed:
added function to check available recipes and print available recipes

notbingsu committed:
coding standards edit

JuneNita committed:
add JUnit Test for 'random', and tagging

JuneNita committed:
update format

JuneNita committed:
Merge pull request #64 from jaredoong/enhancements

Add functionality to add, delete and view list of ingredients user have

woowenjun99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
+
+The sequence diagram below shows how the `WeeklyPlan` component works when the user
+inputs `weekly /add burger 1`:
+![](../docs/UML/WeeklyPlan/AddWeeklyPlanUML.png)
```
The diagram cannot be seen on the developer website. It might be good to review and see if this is the intended effects.

woowenjun99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries --
+include links to the original source as well}
```
Perhaps it is nice to fill this up, or remove it if it is unused.

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started](#setting-up-getting-started)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#ui-component)
+    * [Parser Component](#parser-component)
+    * [Recipe Component](#recipe-component)
+    * [RecipeList Component](#recipelist-component)
+    * [WeeklyPlan Component](#weeklyplan-component)
+    * [Database Component](#database-component)
+* [Implementation](#implementation)
```
Perhaps you could add a numbering format in this table of content and the subsequent subheaders so that it is easier to view, for eg. 
```
- 1. Acknowledgements
- 2. Setting up, getting started
- 3. Design 
    - 3.1 Architecture 
    - 3.2 UI Component
```

woowenjun99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
+
+---
+
+## Setting up, getting started
+
+---
```
As a developer, I would not know how to set up. Perhaps it is good to fill up this set up guide of the developer guide.

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
+
+## Design
+
+* [Architecture](#architecture)
+* [UI Component](#ui-component)
+* [Parser Component](#parser-component)
+* [Recipe Component](#recipe-component)
+* [RecipeList Component](#recipelist-component)
+* [WeeklyPlan Component](#weeklyplan-component)
+* [Database Component](#database-component)
```
I like that you link the subpoints under each subheader, however it may be redundant as you have already added it in the table of content at the start and adding this again would be confusing. Perhaps you could considering either removing this part or the subparts in the table of content at the start.

woowenjun99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
+driven app.
 
-## User Stories
+### User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                     | So that I can ...                                                 |
+|---------|----------|---------------------------------------------------|-------------------------------------------------------------------|
+| v1.0    | user     | add my own recipes to the list                    | refer to them when next time                                      |
+| v1.0    | user     | edit the existing recipe                          |                                                                   |
+| v1.0    | user     | delete a recipe from the list                     | clear the unused recipes                                          |
+| v1.0    | user     | view ingredients of the recipe                    | know what is needed to be prepared                                |
+| v1.0    | user     | list all recipes I have                           | know what I have some idea of what to cook                        |
+| v1.0    | user     | find the recipe that contain specific ingredients | find specific recipe without having to go through the entire list |
+| v1.0    | user     | exit from the program                             |                                                                   |
+| v2.0    | new user | list all the command that can be used             | know what command I can use                                       |
+| v2.0    | user     | add meals I plan to make for the week             | refer to the weekly meals plan next time                          |
+| v2.0    | user     | delete meals I plan to make for the week          | remove some meals from the weekly plan if I change my mind        |
+| v2.0    | user     | categorise recipes using tags                     | group recipes with similar theme together                         |
+| v2.0    | user     | list the recipes by tag                           | list recipes that are under the specific category                 |
 
-## Non-Functional Requirements
+### Non-Functional Requirements
 
 {Give non-functional requirements}
 
-## Glossary
+### Glossary
 
 * *glossary item* - Definition
 
-## Instructions for manual testing
+### Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+{Give instructions on how to do a manual product testing e.g., how to load sample data to be used
+for testing}
```
Perhaps it might be better to give a more detailed description as to how to perform manual testing, glossary, or remove this section completely.

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
+    * If user want to add recipes to a tag, `addRecipeToTag()` is executed to add recipes in
+      to the tag.
+    * If user want to remove recipes to a tag, `removeRecipeFromTag()` is executed to remove recipes
+      from the tag.
```
Very specific explanation of how each step of implementation works, well done.

woowenjun99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                     | So that I can ...                                                 |
+|---------|----------|---------------------------------------------------|-------------------------------------------------------------------|
+| v1.0    | user     | add my own recipes to the list                    | refer to them when next time                                      |
+| v1.0    | user     | edit the existing recipe                          |                                                                   |
+| v1.0    | user     | delete a recipe from the list                     | clear the unused recipes                                          |
+| v1.0    | user     | view ingredients of the recipe                    | know what is needed to be prepared                                |
+| v1.0    | user     | list all recipes I have                           | know what I have some idea of what to cook                        |
+| v1.0    | user     | find the recipe that contain specific ingredients | find specific recipe without having to go through the entire list |
+| v1.0    | user     | exit from the program                             |                                                                   |
+| v2.0    | new user | list all the command that can be used             | know what command I can use                                       |
+| v2.0    | user     | add meals I plan to make for the week             | refer to the weekly meals plan next time                          |
+| v2.0    | user     | delete meals I plan to make for the week          | remove some meals from the weekly plan if I change my mind        |
+| v2.0    | user     | categorise recipes using tags                     | group recipes with similar theme together                         |
+| v2.0    | user     | list the recipes by tag                           | list recipes that are under the specific category                 |
```
I enjoyed how clear the user stories are clear! Good job!

yixuann02 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,312 @@
+* Add a single recipe in 1 line and followed by all the ingredients in next another line after being prompted.
+
+It is implemented through the following steps:
+1. When the user enters an input with the first word being `add`, the input is passed to the `Parser` component.
+2. In `Parser`, the `parseAddRecipe` is executed to identify whether the recipe is an already existing recipe or
+   it's a new recipe that is being added.
```
Perhaps you could provide an example of how an input would look like.

yixuann02 reviewed (commented):
Overall, well done. Do remember to update and add in the diagrams as the diagram cannot be viewed.

arsdorintbp2003 commented:
1. Overall, your UG is well-structured, but it seems that you haven't added your UML diagrams.

arsdorintbp2003 commented:
2. I think the List of components hyperlink in the Design is mentioned twice (on the first page of the UG and inside the Design part, which is redundant. 
![image](https://user-images.githubusercontent.com/88639992/228740279-18edd3ff-5eee-483d-b71a-e5dceede3142.png)

![image](https://user-images.githubusercontent.com/88639992/228740358-43cb8189-1d07-42bc-b9e6-c1e1fbb6aa0e.png)

arsdorintbp2003 commented:
3. I think this part has the wrong format since all your other explanations use numbering or bullet, instead of "{}". ![image](https://user-images.githubusercontent.com/88639992/228740745-7cba2409-12df-4c93-bf09-b2beeb451bc6.png)

arsdorintbp2003 commented:
4. Another format error, your Design and Implementation all got the list of components/ feature inside the part, but Appendix don't have 
![image](https://user-images.githubusercontent.com/88639992/228741663-7af3f1b7-bdc9-4a81-8f4a-726f43205d8c.png)

arsdorintbp2003 commented:
5. I think you could add more examples of how your input looks like (e.g you can paste the input from your .jar file running
![image](https://user-images.githubusercontent.com/88639992/228741927-411289e3-1463-4bd1-9a84-a9b46634a3d8.png)

gurmankalkat committed:
fix delete syntax bug

gurmankalkat committed:
add sequence diagram for delete function

notbingsu committed:
no message

gurmankalkat committed:
add to help command

notbingsu committed:
no message

gurmankalkat committed:
fix spacing error with delete command

TopGun2001 committed:
Edited the help commands. Updated user guide.

notbingsu committed:
Merge branch 'master' of https://github.com/notbingsu/tp

TopGun2001 committed:
.

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-BeforeSthGoneWrong

notbingsu committed:
no message

jaredoong committed:
Merge pull request #70 from notbingsu/master

add function allowing users to view available ingredients

jaredoong committed:
Merge pull request #71 from JuneNita/JuneNita-BeforeSthGoneWrong

Add JUnit tag and random

jaredoong committed:
Merge branch 'master' into master

jaredoong committed:
Merge pull request #69 from TopGun2001/branch-level-1

Edited the help commands. Updated user guide.

jaredoong committed:
Add functionality improve user experience

User can mark recipes in the weekly plan as done, and the amount of ingredients are automatically reduced

jaredoong committed:
Fix minor merge conflicts

notbingsu committed:
Merge pull request #74 from jaredoong/enhancements

Add function to improve user experience

gurmankalkat committed:
fix formatting of returned string from parseDeleteRecipe

gurmankalkat committed:
fix merge conflicts

gurmankalkat committed:
Merge branch 'master' into master

jaredoong committed:
Merge pull request #68 from gurmankalkat/master

fix delete syntax bug, add sequence diagram, update help command

jaredoong committed:
Update parsing of user input

Allow all commands to be case insensitive instead.
Edit parsing of `weekly /clear` command to ensure it gets executed.

* Closes #114
* Closes #98

theopin commented on src/main/java/seedu/meal360/RecipeList.java:
```diff
@@ -0,0 +1,145 @@
+            tagRecipes = this.tags.get(filter);
+            hasNoRecipeInTheTag = tagRecipes == null;
+            hasNoRecipeToReturn = filteredRecipeList.size() == 0;
+
+            if (hasNoRecipeInTheTag) {
+                throw new NullPointerException("There is nothing to list.");
+            }
```
Perhaps consider making this message as a constant instead of leaving it as a string literal

theopin commented on src/main/java/seedu/meal360/RecipeList.java:
```diff
@@ -0,0 +1,145 @@
+        this.tags.get(filters[0].trim()).forEach((recipe, dummy) -> filteredRecipeList.add(recipe));
+
+        for (String filter : filters) {
+            filter = filter.trim();
```
try to avoid reusing variables. Perhaps consider renaming it into a different variable name?

theopin commented on src/main/java/seedu/meal360/Parser.java:
```diff
@@ -0,0 +1,641 @@
+            commandString.append(" ").append(inputs[i]);
+        }
+
+        isAddTag = commandString.indexOf(">>") == -1 && commandString.indexOf("<<") != -1;
+        isRemoveTag = commandString.indexOf("<<") == -1 && commandString.indexOf(">>") != -1;
+
+        if (!(isAddTag || isRemoveTag)) {
+            throw new IllegalArgumentException("Please enter the command in the correct format.");
+        } else if (isAddTag) {
+            tag = parseAddRecipeTag(commandString.toString(), recipeList);
+        } else if (isRemoveTag) {
+            tag = parseRemoveRecipeTag(commandString.toString(), recipeList);
+        } else {
+            throw new IllegalArgumentException("Invalid command.");
```
Same issue of magic literals. Try to convert them into constants so that it can help to enhance readability of code

theopin commented on src/main/java/seedu/meal360/Parser.java:
```diff
@@ -0,0 +1,641 @@
+            recipeToDelete = recipeToDelete.trim();
+            if (recipeToDelete.equals("all")) {
+                String allRecipes = "";
+                int index = 0;
+                while (recipeList.size() != 0) {
+                    allRecipes += recipeList.deleteRecipe(index).getName() + ", ";
+                }
+                allRecipes = allRecipes.substring(0, allRecipes.length() - 2);
+                return allRecipes;
+            } else {
+                int recipeIndex = 0;
+                for (Recipe recipe : recipeList) {
+                    // find index of recipe we want to delete
+                    if (recipe.getName().equals(recipeToDelete)) {
+                        break;
+                    }
+                    recipeIndex++;
+                }
+                return recipeList.deleteRecipe(recipeIndex).getName();
+            }
+            // user inputted index of recipe in list
+        } else {
+            // deleting a range of recipes
+            if (input[1].length() >= 3) {
+                String rangeRecipes = "";
+                int startIndex = Integer.parseInt(input[1].charAt(0) + "");
+                int endIndex = Integer.parseInt(input[1].charAt(2) + "");
+                startIndex -= 1;
+                endIndex -= 1;
+                int newSize = recipeList.size() - ((endIndex - startIndex) + 1);
+                while (recipeList.size() != newSize) {
+                    rangeRecipes += recipeList.deleteRecipe(startIndex).getName() + ", ";
+                }
+                rangeRecipes = String.valueOf(new StringBuilder(rangeRecipes.substring(0, rangeRecipes.length() - 2)));
+                return rangeRecipes;
+            } else {
+                int recipeIndex = Integer.parseInt(input[1]);
+                // need to subtract 1 since list is 1-based
+                return recipeList.deleteRecipe(recipeIndex - 1).getName();
+            }
```
Method seems quite big. Perhaps consider finding related behaviour and abstracting them into various functions

theopin commented on src/main/java/seedu/meal360/Ui.java:
```diff
@@ -0,0 +1,141 @@
+        } else {
+            printMessage("Here are your weekly ingredients:");
+            weeklyPlan.forEach((recipe, count) -> {
+                Recipe currRecipe = recipeList.findByName(recipe.toString().trim());
+                for (String ingredient : currRecipe.getIngredients().keySet()) {
```
Seems like a lot of actions happening here. Perhaps consider breaking down certain multiple functions into separate variables for easier readability (eg. String sampleName = urrRecipe.getIngredients().keySet())

theopin commented on src/main/java/seedu/meal360/Parser.java:
```diff
@@ -0,0 +1,641 @@
+                    System.out.println(ui.formatMessage(toPrint));
+                    ui.printSeparator();
+                    break;
+                }
+                count++;
+            }
+            System.out.println("Please enter the new ingredient:");
+            String line = userInput.nextLine();
+            String command = line.replaceAll("\\s+", " ");
+            int indexOfEqual = command.indexOf("=");
+            String newIngredientName = command.substring(0, indexOfEqual);
+            int newIngredientQuantity = Integer.parseInt(command.substring(indexOfEqual + 1));
+            recipeToEdit.getIngredients().remove(ingredientToRemove);
+            recipeToEdit.getIngredients().put(newIngredientName, newIngredientQuantity);
+            recipeList.editRecipe(recipeToEdit, recipeToEdit.getIngredients());
+        } else if (index == 3) {
+            HashMap<String, Integer> newIngredientList = recipeToEdit.getIngredients();
+            System.out.println("These are the current ingredients:");
+            ui.printSeparator();
+            Recipe recipe = parseViewRecipe(recipeName, recipeList);
+            ui.printRecipe(recipe);
+            ui.printSeparator();
+            System.out.println(
+                    "Please Enter Additional Ingredients & Quantity (Enter done when complete): ");
+            while (true) {
+                String line = userInput.nextLine();
+                if (line.equals("done")) {
+                    ui.printSeparator();
+                    break;
+                } else {
+                    String[] command = line.trim().split(" ");
+                    ingredients = parseIngredientName(command);
+                    newIngredientList.putAll(ingredients);
+                }
+            }
+            recipeList.editRecipe(recipeToEdit, newIngredientList);
+        }
+
+        return recipeToEdit;
+    }
```
Method is very large, consider taking out common behaviour and refactor to other methods for easier maintainability

theopin commented on src/main/java/seedu/meal360/Parser.java:
```diff
@@ -0,0 +1,641 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+public class Parser {
+
```
Seems like the parser class is doing too much (ie. handling all forms of commands within itself). Perhaps consider splitting parser into a few classes for handling each important group of command so that it reduces the functionality of parser and each sub class can be independently handled easily

theopin commented on src/main/java/seedu/meal360/storage/Database.java:
```diff
@@ -0,0 +1,189 @@
+        userIngredientsJsonDatabaseHelper.saveDatabase(userIngredients);
+    }
+
+    public RecipeList defaultRecipeList() {
```
name of method should be an action eg. createDefaultRecipeList

theopin commented on src/main/java/seedu/meal360/storage/Database.java:
```diff
@@ -0,0 +1,189 @@
+
+    public RecipeList defaultRecipeList() {
+        RecipeList defaultRecipeList = new RecipeList();
+        defaultRecipeList.add(new Recipe("Chicken Rice", new HashMap<>() {
+            {
+                put("Chicken", 1);
+                put("Rice", 1);
+                put("Salt", 1);
+                put("Pepper", 1);
+            }
```
Perhaps consider refactoring such hashmaps as constants to help reduce the LOC of this method and also give greater clarity as to its purpose: eg i am adding new recipe called CHICKEN_RICE with CHICKEN_RICE_INGREDIENTS (the hashmap)

theopin commented on src/main/java/seedu/meal360/Parser.java:
```diff
@@ -0,0 +1,641 @@
+    }
+
+    public HashMap<String, Integer> parseIngredientName(String[] command) {
+        HashMap<String, Integer> ingredients = new HashMap<>();
+        int flag = 0;
+        StringBuilder add = null;
+        String ingredientName;
+        int ingredientQuantity;
+        int indexOfEqual;
+
+        for (String s : command) {
+            try {
+                indexOfEqual = s.indexOf("=");
+                if (indexOfEqual == -1) {
+                    if (add != null) {
+                        add.append(" ").append(s);
+                    } else {
+                        add = new StringBuilder(s);
+                    }
+                    flag++;
+
+                } else {
+                    if (flag > 0) {
+                        indexOfEqual = s.indexOf("=");
+                        ingredientName = (s.substring(0, indexOfEqual));
+                        ingredientName = add + " " + ingredientName;
+                        ingredientQuantity = Integer.parseInt(s.substring(indexOfEqual + 1));
+                        flag = 0;
+                        add = null;
+                    } else {
+                        ingredientName = s.substring(0, indexOfEqual);
+                        ingredientQuantity = Integer.parseInt(s.substring(indexOfEqual + 1));
+                    }
+
+                    ingredients.put(ingredientName, ingredientQuantity);
```
Method seems to be very nested, perhaps try to reduce the number of nests if it does not severely affect your code logic?

theopin commented on src/main/java/seedu/meal360/Meal360.java:
```diff
@@ -0,0 +1,297 @@
+        } else if (command[0].equals("weeklyingredients")) {
+            ui.printSeparator();
+            ui.printWeeklyIngredients(weeklyPlan, recipeList);
+            ui.printSeparator();
+        } else if (command[0].equals("weeklyplan")) {
+            ui.printSeparator();
+            ui.printWeeklyPlan(weeklyPlan);
+            ui.printSeparator();
+        } else if (command[0].equals("add_i")) {
+            ui.printSeparator();
+            try {
+                parser.parseAddUserIngredients(command, userIngredients);
+                ui.printMessage("Ingredient successfully added!");
+            } catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+                ui.printMessage(e.getMessage());
+            }
+            ui.printSeparator();
+        } else if (command[0].equals("del_i")) {
+            ui.printSeparator();
+            try {
+                parser.parseDeleteUserIngredients(command, userIngredients);
+                ui.printMessage("Ingredients successfully deleted!");
+            } catch (IllegalArgumentException | IndexOutOfBoundsException | IngredientNotFoundException e) {
+                ui.printMessage(e.getMessage());
+            }
+            ui.printSeparator();
+        } else if (command[0].equals("view_ingredients")) {
+            ui.printSeparator();
+            ui.printUserIngredients(userIngredients);
+            ui.printSeparator();
+        } else if (command[0].equals("help")) {
+            ui.printSeparator();
+            ui.printHelp();
+            ui.printSeparator();
+        } else {
+            ui.printSeparator();
+            ui.printMessage("I'm sorry, but I don't know what that means :-(");
+            ui.printSeparator();
+        }
+    }
```
method seems to be very long. Do consider grouping related behaviour and refactoring it into smaller methods to reduce its LOC and to make it more concise

theopin reviewed (commented):
Left some comments to improve the quality of your code. All the best for your implementation!

jaredoong committed:
Bug fixes and minor updates to commands and user docs

Add protection against overly large/small number, quantity for number of
times each recipe can be included in weekly plan is capped at 1000.
Add clearer error message for when integer provided for `view` function
results in overflow.
Update `weekly /add` in user documentation to be more specific.
Update parsing of user input to be non case-sensitive.

* Closes #100
* Closes #95
* Closes #87
* Closes #86

jaredoong committed:
Update JUnit test and user docs

Fix bugs in functions regarding adding and deleting recipes from weekly
plan.
Update JUnit test and user docs for the following few commands:
* `weekly /add`
* `weekly /multiadd`
* `weekly /delete`
* `weekly /multidelete`
* 'view`

jaredoong committed:
Fix random errors with checks

jaredoong committed:
Update JUnit and user docs

Add Junit for commands to add, delete and view list of user ingredients.
Fix bug where `bye` command not working with trailing or leading
whitespaces.
Update user docs to explanation for default list of recipes.

* Closes #104
* Closes #82
* Closes # 73

jaredoong committed:
Merge pull request #120 from jaredoong/JUnit-and-docs

JUnit and docs

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-BeforeSthGoneWrong

JuneNita committed:
Add junenita-PPP

jaredoong committed:
Add Jared-PPP

Add PPP for Jared and removed the dummy johndoe.md.

JuneNita committed:
update the arrow of the activation bar

jaredoong committed:
Update DG

Add Architecture and user stories

jaredoong committed:
Merge pull request #121 from JuneNita/JuneNita-BeforeSthGoneWrong

Add JuneNita-PPP

JuneNita committed:
update 'tag' UML pic

jaredoong committed:
Merge pull request #122 from JuneNita/JuneNita-updateUML

fix issue#119 ('tag' UML)

JuneNita committed:
Merge pull request #123 from jaredoong/PPP-Jared

Add PPP

gurmankalkat committed:
fix errors with delete feature & add PPP

gurmankalkat committed:
update PPP

gurmankalkat committed:
Merge branch 'master' into master

TopGun2001 committed:
...

JuneNita committed:
update link on aboutUs

JuneNita committed:
update Tag UI message

JuneNita committed:
modifiy tag test

TopGun2001 committed:
Created PPP. Updated the code based on practical dry run bugs.

TopGun2001 committed:
Merge pull request #126 from TopGun2001/branch-level-1

Branch level 1

JuneNita committed:
Merge branch 'update-tag' into JuneNita-BeforeSthGoneWrong

JuneNita committed:
fix 'random' exception

gurmankalkat committed:
fix formatting for error line

gurmankalkat committed:
Merge branch 'master' of https://github.com/gurmankalkat/tp

gurmankalkat committed:
Merge pull request #124 from gurmankalkat/master

fix delete feature errors & update PPP

gurmankalkat committed:
Merge pull request #125 from JuneNita/update-tag

modified 'tag' return message

gurmankalkat committed:
Merge pull request #127 from JuneNita/JuneNita-BeforeSthGoneWrong

fix 'random' exception handler

TopGun2001 committed:
Changed PPP file name

TopGun2001 committed:
Merge https://github.com/AY2223S2-CS2113-F10-3/tp into branch-level-1

TopGun2001 committed:
Changed ppp file name.

TopGun2001 committed:
New changes.

TopGun2001 committed:
Merge pull request #128 from TopGun2001/branch-level-1

Branch level 1

jaredoong committed:
Update DG

Add Architecture to show overall struture of Meal360.
Add 3 class diagrams to show the 3 subsystems of Meal360 which are the
recipes, ingredients and weekly meal plan.
Update implementation of ingredient and weekly plan related features.

jaredoong committed:
Merge branch 'master' of github.com:AY2223S2-CS2113-F10-3/tp into PPP-Jared

jaredoong committed:
Add manual testing to DG

Fix minor display bugs

jaredoong committed:
Updated expected.txt to pass check

jaredoong committed:
Fix sequence diagram

Fix return arrow not at end of activation bar for EditWeeklyPlan and MarkDoneWeeklyPlan

jaredoong committed:
Update DG and PPP to reflect new changes

Fixed minor inconsistencies in DG

jaredoong committed:
Update Architecture diagram

Simplified diagram to be neater

JuneNita committed:
Merge pull request #129 from jaredoong/PPP-Jared

Update DG to match v2.1

jaredoong committed:
Update README.md

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-BeforeSthGoneWrong

JuneNita committed:
update file path

JuneNita committed:
update file path

JuneNita committed:
Delete docs/UML/Implementation/randomFunction directory

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-BeforeSthGoneWrong

notbingsu committed:
Update Ui class diagram

notbingsu committed:
Rename Untitled Diagram.drawio to Ui class diagram

notbingsu committed:
update Ui component in developer's guide

notbingsu committed:
Added UiClassDiagram.drawio.png

notbingsu committed:
Ui component DG update

notbingsu committed:
Merge branch 'master' of https://github.com/notbingsu/tp

notbingsu committed:
Added Ui_sequencediagram.drawio

notbingsu committed:
Added commandList.drawio.png

notbingsu committed:
Updated Ui component DG with UML

notbingsu committed:
updated UI component in DG

notbingsu committed:
Added parseDate.drawio

notbingsu committed:
Update parseDate.drawio

notbingsu committed:
Added parseDate.drawio.png

notbingsu committed:
updated Parser component for DG

notbingsu committed:
Update parseDate.drawio

notbingsu committed:
Update parseDate.drawio

notbingsu committed:
fixed issue where it is possible to have empty string as ingredient name

notbingsu committed:
checkstyle update

JuneNita committed:
update exception

JuneNita committed:
Merge branch 'update-exception' into JuneNita-BeforeSthGoneWrong

JuneNita committed:
Merge branch 'JuneNita-BeforeSthGoneWrong' into documentation

JuneNita committed:
edit UG, DG, portfolio

jaredoong committed:
Merge pull request #130 from notbingsu/master

Added architecture and description for Ui and Parser components in DG; added checks for add recipe method

JuneNita committed:
add JavaDoc

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into documentation

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into JuneNita-BeforeSthGoneWrong

JuneNita committed:
edit ppp

JuneNita committed:
edit DG

jaredoong committed:
Merge pull request #133 from JuneNita/JuneNita-BeforeSthGoneWrong

update exception and add javadoc

JuneNita committed:
Merge branch 'master' into documentation

JuneNita committed:
Merge pull request #134 from JuneNita/documentation

edit UG, DG and PPP

jaredoong committed:
Update DG

- Add class diagrams for Meal360, RecipeList, IngredientList, Database,
  and WeeklyPlan components.
- Update personal PPP
- Fix AboutUs.md table

jaredoong committed:
Fix merge conflicts

jaredoong committed:
Fix filepath for DG images

jaredoong committed:
Refactor adding,deleting of ingredients, and marking recipe as done

jaredoong committed:
Update CG to reflect refactored code

gurmankalkat committed:
fix DeleteFunction diagram in DG

jaredoong committed:
Merge pull request #136 from gurmankalkat/master

fix DeleteFunction diagram in DG

gurmankalkat committed:
Merge pull request #135 from jaredoong/PPP-Jared

Update DG

gurmankalkat committed:
fix bug with adding recipe & update PPP

gurmankalkat committed:
clean up code

gurmankalkat committed:
add more clarification to recipe error message

gurmankalkat committed:
Merge branch 'master' into master

gurmankalkat committed:
add to DG

gurmankalkat committed:
Merge branch 'master' of https://github.com/gurmankalkat/tp

TopGun2001 committed:
Merge pull request #137 from gurmankalkat/master

fix bug with adding recipe, update PPP, update DG & UG

TopGun2001 committed:
Added UML diagrams for add and edit functions. Updated User guide. Updated Developer Guide, added Parser component diagram to Parser class in DG.

TopGun2001 committed:
Merge branch 'master' into branch-level-1

TopGun2001 committed:
Updated the HELP commands.

TopGun2001 committed:
Updated my PPP.

jaredoong committed:
Update UserGuide.md

Add explanation that current session saved only when program exits properly

jaredoong committed:
Update UserGuide.md

Add exiting of program to command summary

TopGun2001 committed:
Added documentation for methods.

TopGun2001 committed:
Merge branch 'master' into branch-level-1

TopGun2001 committed:
Added documentation for a few methods.

TopGun2001 committed:
Resolved merge conflict.

TopGun2001 committed:
.

TopGun2001 committed:
Merge pull request #139 from TopGun2001/branch-level-1

Added UML diagrams for add and edit functions. Updated User guide. Up…

TopGun2001 committed:
Revert "Added UML diagrams for add and edit functions. Updated User guide. Up…"

TopGun2001 committed:
Merge pull request #140 from AY2223S2-CS2113-F10-3/revert-139-branch-level-1

Revert "Added UML diagrams for add and edit functions. Updated User guide. Up…"

TopGun2001 committed:
Updated user guide.

TopGun2001 committed:
Merge branch 'master' into branch-level-1

TopGun2001 committed:
FINAL CHANGES.

TopGun2001 committed:
FINAL CHANGES.

TopGun2001 committed:
Last Changes to UG.

TopGun2001 committed:
Last Changes made to UG.

notbingsu committed:
update help function to include "available" function

TopGun2001 committed:
Added documentation and made changes to user guide.

TopGun2001 committed:
Merge pull request #143 from TopGun2001/branch-level-1

Branch level 1

TopGun2001 committed:
Merge pull request #142 from notbingsu/master

update help function to include "available" function

TopGun2001 committed:
Updated DG, UG, PPP.

TopGun2001 committed:
Update topgun2001.md

notbingsu committed:
Added availableSequenceDiagram.drawio

TopGun2001 committed:
Merge pull request #144 from TopGun2001/branch-level-1

DG,UG,PPP

TopGun2001 committed:
DG

TopGun2001 committed:
Merge https://github.com/AY2223S2-CS2113-F10-3/tp into last-branch

TopGun2001 committed:
Merge pull request #145 from TopGun2001/last-branch

Last branch

notbingsu committed:
Added availableSequenceDiagram.drawio.png

notbingsu committed:
Delete Untitled Diagram.drawio

notbingsu committed:
Added availableSequenceDiagram.drawio

notbingsu committed:
Update availableSequenceDiagram.drawio

notbingsu committed:
Update availableSequenceDiagram.drawio

notbingsu committed:
update Implementation in DG with available function

notbingsu committed:
Merge branch 'master' of https://github.com/notbingsu/tp

jaredoong committed:
Merge pull request #147 from notbingsu/master

Update DG

jaredoong committed:
Update DG

Update recipe related subsystem diagram to reflect available recipes function

notbingsu committed:
updated notbingsu PPP and link in AboutUs.md

notbingsu committed:
Merge pull request #148 from jaredoong/DG-update

Update DG

notbingsu committed:
no message

JuneNita committed:
Merge pull request #149 from notbingsu/master

updated notbingsu PPP and link in AboutUs.md

gurmankalkat committed:
add more exception handling & testing for that, update PPP & DG

gurmankalkat committed:
Merge branch 'master' into master

jaredoong committed:
Merge pull request #150 from gurmankalkat/master

add more exception handling & testing for that, update PPP & DG

jaredoong committed:
Update author tgas

jaredoong committed:
Merge branch 'master' into DG-Update

jaredoong committed:
Merge pull request #151 from jaredoong/DG-Update

Update author tags

jaredoong committed:
Fix visual bugs in UG and DG

Format code to allow UG and DG to load properly

jaredoong committed:
Merge branch 'DG-Update' of github.com:jaredoong/tp into DG-Update

jaredoong committed:
Merge branch 'master' of github.com:AY2223S2-CS2113-F10-3/tp into DG-Update

jaredoong committed:
Fix visual bugs in UG and DG

Format code to allow UG and DG to load properly

jaredoong committed:
Merge pull request #152 from jaredoong/DG-Update

Fix visual bugs in UG and DG

TopGun2001 committed:
Final edits made to add function. Updated help commands.

TopGun2001 committed:
Merge pull request #153 from TopGun2001/last-branch

Final edits made to add function. Updated help commands.

gurmankalkat committed:
update PPP & DG

jaredoong committed:
Merge pull request #154 from gurmankalkat/master

update PPP & DG

jaredoong committed:
Update DeveloperGuide.md

Fix numbering of subsystem diagrams not showing up properly in github pages

jaredoong committed:
Fix weekly planner bug

Fix bug where weekly /done not being read properly

jaredoong committed:
Merge pull request #155 from jaredoong/bugfixes

Fix weekly planner bug

JuneNita committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-F10-3/tp into documentation

JuneNita committed:
remove case-sensitive

JuneNita committed:
Merge pull request #156 from JuneNita/documentation

edit case-sensitivity in UG

gurmankalkat committed:
fix help command for delete

gurmankalkat committed:
Merge branch 'AY2223S2-CS2113-F10-3:master' into master

gurmankalkat committed:
Merge pull request #157 from gurmankalkat/master

fix help command for deleting recipe

jaredoong committed:
Update error messages

Edit error messages to give users better idea on the error

jaredoong committed:
Add graceful force exit

Add catching of Ctrl-C/Ctrl-Z and attempts to save data before exiting

jaredoong committed:
Update PPP

jaredoong committed:
Fix PPP visual bug

jaredoong committed:
Update testing to pass checks

jaredoong committed:
Merge pull request #158 from jaredoong/bugfixes

Bugfixes
